src/components/ClientSideLoggingToolMenu/index.native.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useState} from 'react';
+import {Alert} from 'react-native';
+import RNFetchBlob from 'react-native-blob-util';
+import {withOnyx} from 'react-native-onyx';
+import Share from 'react-native-share';
+import * as Console from '@libs/actions/Console';
+import {parseStringifyMessages} from '@libs/Console';
+import type {Log} from '@libs/Console';
+import getOperatingSystem from '@libs/getOperatingSystem';
+import localFileCreate from '@libs/localFileCreate';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import BaseClientSideLoggingToolMenu from './BaseClientSideLoggingToolMenu';
+import type {ClientSideLoggingToolMenuOnyxProps, ClientSideLoggingToolProps} from './types';
+
+function ClientSideLoggingToolMenu({shouldStoreLogs, capturedLogs}: ClientSideLoggingToolProps) {
+    const [file, setFile] = useState<{path: string; newFileName: string; size: number}>();
+
+    const onToggle = () => {
+        if (!shouldStoreLogs) {
+            Console.setShouldStoreLogs(true);
+        } else {
+            if (capturedLogs) {
```

Comment:
these could be simplified with early returns on all the edge cases (like no logs captured)